test/ci-cd/scripts/comfyui/run_in_cluster.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/sh
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+#
+# GKE In-Pod Runner
+#
+# Description:
+#   This script automates running a test inside a temporary GKE pod.
+#   It is configured entirely via environment variables.
+#
+# Required Environment Variables:
+#   cluster_name, cluster_region, cluster_project_id, comfyui_kubernetes_namespace, comfyui_endpoints_hostname,
+#
+# ==============================================================================
+
+# --- Script Configuration ---
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source /workspace/build.env
+if [ "${DEBUG,,}" == "true" ]; then
+  set -o xtrace
+fi
+
+# Check for all required environment variables
+if [ -z "$cluster_name" ] || [ -z "$cluster_region" ] || [ -z "$cluster_project_id" ] || \
+   [ -z "$comfyui_kubernetes_namespace" ] || [ -z "$comfyui_endpoints_hostname" ] || [ -z "$WORKFLOWS_DIR" ] || \
+   [ -z "$TEST_DIR" ]; then
+    echo "Error: One or more required environment variables are not set." >&2
+    echo "Required: cluster_name, cluster_region, cluster_project_id, comfyui_kubernetes_namespace, HEADLESS_COMFYUI_SERVICE, WORKFLOWS_DIR, TEST_DIR" >&2
+    exit 1
+fi
+
+# --- Variables ---
+# Use a unique name for the pod for concurrent runs.
+POD_NAME="comfyui-client"
+printenv
+# --- verify each file has a test file ---
+check_files() {
+  for file in "$WORKFLOWS_DIR"/*; do
+    if [[ ! -f "$TEST_DIR/workflows/$(basename "$file")" ]]; then
+      echo "Error: $(basename "$file") not found in $TEST_DIR." >&2
+      exit 1
+    fi
+  done
+}
+# --- Cleanup Function ---
+cleanup() {
+  echo "--- Cleaning up pod: $POD_NAME ---"
+  kubectl delete pod "$POD_NAME" --namespace="$comfyui_kubernetes_namespace" --ignore-not-found=true || true
+}
+trap cleanup EXIT
+
+
+# 1. Get GKE cluster credentials
+echo "--- Getting GKE credentials for project '$cluster_project_id' ---"
+gcloud container clusters get-credentials "$cluster_name" --region "$cluster_region" --project "$cluster_project_id" --dns-endpoint
```

Comment:
Use the `${cluster_credentials_command}` to get credentials as it might change depending on the configuration of the environment.